src/status_im2/contexts/chat/messages/link_preview/events.cljs

```diff
@@ -89,31 +17,86 @@
    :link-previews-cache
    (assoc (get multiaccount :link-previews-cache {}) site data)))
 
-(defn cache-community-preview-data
-  [{:keys [id] :as community}]
-  (re-frame/dispatch [::cache-link-preview-data
-                      (community-link id)
-                      community]))
+(rf/defn load-link-preview-data
+  {:events [:chat.ui/load-link-preview-data]}
+  [{{:keys [multiaccount]} :db :as cofx} link]
+  (if (get-in multiaccount [:link-previews-cache link])
+    {:db (:db cofx)}
+    {:json-rpc/call [{:method     "wakuext_getLinkPreviewData"
+                      :params     [link]
+                      :on-success #(rf/dispatch [::cache-link-preview-data link %])
+                      :on-error   #(rf/dispatch [::cache-link-preview-data link
+                                                 {:error (str "Can't get preview data for " link)}])}]}))
 
 (rf/defn should-suggest-link-preview
-  {:events [::should-suggest-link-preview]}
+  {:events [:chat.ui/should-suggest-link-preview]}
   [{:keys [db] :as cofx} enabled?]
   (multiaccounts.update/multiaccount-update
    cofx
    :link-preview-request-enabled
    (boolean enabled?)
    {}))
 
+(rf/defn handle-community-resolved
+  {:events [::community-resolved]}
+  [{:keys [db] :as cofx} community-id community]
+  (rf/merge cofx
+            (cond-> {:db (update db :communities/resolve-community-info dissoc community-id)}
+
+              (some? community)
+              (assoc :dispatch [::cache-link-preview-data (community-link community-id) community]))
+            (models.communities/handle-community community)))
+
+(rf/defn handle-community-failed-to-resolve
+  {:events [::community-failed-to-resolve]}
+  [{:keys [db]} community-id]
+  {:db (update db :communities/resolve-community-info dissoc community-id)})
+
+(rf/defn resolve-community-info
+  {:events [:chat.ui/resolve-community-info]}
+  [{:keys [db]} community-id]
+  {:db            (assoc-in db [:communities/resolve-community-info community-id] true)
+   :json-rpc/call [{:method     "wakuext_requestCommunityInfoFromMailserver"
+                    :params     [community-id]
+                    :on-success #(rf/dispatch [::community-resolved community-id %])
+                    :on-error   #(do
+                                   (rf/dispatch [::community-failed-to-resolve community-id])
+                                   (log/error "Failed to request community info from mailserver"))}]})
+
+(rf/defn save-link-preview-whitelist
+  {:events [::link-preview-whitelist-received]}
```

Comment:
we decided to do not use namespaced events https://github.com/status-im/status-mobile/blob/develop/doc/new-guidelines.md#subscription-names-and-event-names

Reply:
for all cases in this namespace

Reply:
Even for something that shouldn't be used outside the namespace? Interesting but okay.

Reply:
yeah, just simpler to do not use it at all, less questions and confusions